java/system-test/src/main/java/sleeper/systemtest/ingest/RunWriteRandomDataTaskOnECS.java

```diff
@@ -97,19 +96,7 @@ public List<RunTaskResult> run() {
                 .withPlatformVersion(systemTestProperties.get(FARGATE_VERSION));
 
         List<RunTaskResult> results = new ArrayList<>();
-        for (int i = 0; i < systemTestProperties.getInt(NUMBER_OF_WRITERS); i++) {
-            // Rate limit for Fargate tasks is 100 burst, 20 sustained:
-            // https://docs.aws.amazon.com/AmazonECS/latest/userguide/throttling.html
-            if (i > 0) {
-                sleepForSustainedRatePerSecond(10);
-            }
-            RunTaskResult result = ecsClient.runTask(runTaskRequest);
-            List<Failure> failures = result.getFailures();
-            if (!failures.isEmpty()) {
-                throw new ECSFailureException("Failures running task " + i + ": " + failures);
-            }
-            results.add(result);
-        }
+        RunECSTasks.runTasksOrThrow(ecsClient, runTaskRequest, systemTestProperties.getInt(NUMBER_OF_WRITERS), () -> true, results::add);
```

Comment:
It looks like this will abort running tasks if more than one request to ECS is needed, since the checkAbort supplier will return true.

It will also return successfully if failures are returned from a request, where before it would throw an ECSFailureException.

Reply:
Done, thanks